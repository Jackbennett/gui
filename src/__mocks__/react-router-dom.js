import React from 'react';
import { compose, setDisplayName } from 'recompose';
import * as rrd from 'react-router-dom';
// Just render plain div with its children
// eslint-disable-next-line no-import-assign
rrd.BrowserRouter = compose(setDisplayName('BrowserRouter'))(({ children }) => <div>{children}</div>);

module.exports = rrd;
