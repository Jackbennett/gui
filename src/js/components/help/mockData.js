export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
