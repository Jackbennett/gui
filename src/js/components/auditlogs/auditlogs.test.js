import React from 'react';
import { prettyDOM } from '@testing-library/dom';
import { act, render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import AuditLogs from './auditlogs';
import { defaultState, mockDate, undefineds } from '../../../../tests/mockData';

const mockStore = configureStore([thunk]);

describe('Auditlogs Component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({ ...defaultState });
    const _Date = Date;
    global.Date = jest.fn(() => mockDate);
    global.Date.parse = _Date.parse;
    global.Date.now = _Date.now;
    global.Date.toISOString = _Date.toISOString;
    global.Date.UTC = _Date.UTC;
    global.Date.getUTCFullYear = _Date.getUTCFullYear;
    global.Date.getUTCMonth = _Date.getUTCMonth;
    global.Date.getUTCDate = _Date.getUTCDate;
  });

  it('renders correctly', async () => {
    const { baseElement } = render(
      <MemoryRouter>
        <Provider store={store}>
          <AuditLogs />
        </Provider>
      </MemoryRouter>
    );
    const view = prettyDOM(baseElement.firstChild, 100000, { highlight: false })
      .replace(/id="mui-[0-9]*"/g, '')
      .replace(/aria-labelledby="(mui-[0-9]* *)*"/g, '')
      .replace(/\\/g, '');
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });

  it('works as expected', async () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <AuditLogs />
        </Provider>
      </MemoryRouter>
    );
    userEvent.click(screen.getByText(/last 7 days/i));
    userEvent.click(screen.getByText(/clear filter/i));
    userEvent.click(screen.getByRole('button', { name: /Download results as csv/i }));
    userEvent.click(screen.getByText(/open_terminal/i));
  });

  it('allows navigating by url as expected', async () => {
    const ui = (
      <MemoryRouter initialEntries={['/auditlog?start_date=2020-01-01T00:00:00.000Z']}>
        <Provider store={store}>
          <AuditLogs />
        </Provider>
      </MemoryRouter>
    );
    const { rerender } = render(ui);
    await act(() => waitFor(() => rerender(ui)));
    userEvent.click(screen.getByText(/clear filter/i));
  });
});
