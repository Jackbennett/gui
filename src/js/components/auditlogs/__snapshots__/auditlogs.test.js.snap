// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auditlogs Component renders correctly 1`] = `
<div
  className="fadeIn margin-left margin-top-large flexbox column"
  style={
    Object {
      "marginRight": "5%",
    }
  }
>
  <h3>
    Audit log
  </h3>
  <div
    className="auditlogs-filters margin-bottom"
  >
    <div
      aria-expanded={false}
      aria-owns={null}
      className="MuiAutocomplete-root MuiAutocomplete-hasClearIcon"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="combobox"
      style={
        Object {
          "maxWidth": 250,
        }
      }
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
          data-shrink={true}
          htmlFor="audit-log-user-selection"
          id="audit-log-user-selection-label"
        >
          Filter by user
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
          onClick={[Function]}
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-invalid={false}
            autoCapitalize="none"
            autoComplete="off"
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd"
            disabled={false}
            id="audit-log-user-selection"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            placeholder="Select a user"
            required={false}
            spellCheck="false"
            type="text"
            value=""
          />
          <div
            className="MuiAutocomplete-endAdornment"
          >
            <button
              aria-label="Clear"
              className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Clear"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-expanded={false}
      aria-owns={null}
      className="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="combobox"
      style={
        Object {
          "marginLeft": 7.5,
        }
      }
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
          data-shrink={true}
          htmlFor="audit-log-type-selection"
          id="audit-log-type-selection-label"
        >
          Filter by change
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
          onClick={[Function]}
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-invalid={false}
            autoCapitalize="none"
            autoComplete="off"
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd"
            disabled={false}
            id="audit-log-type-selection"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            placeholder="Type"
            required={false}
            spellCheck="false"
            type="text"
            value=""
          />
          <div
            className="MuiAutocomplete-endAdornment"
          >
            <button
              aria-label="Clear"
              className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Clear"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="Open"
              className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Open"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-expanded={false}
      aria-owns={null}
      className="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="combobox"
      style={
        Object {
          "marginRight": 15,
          "marginTop": 16,
        }
      }
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiAutocomplete-inputRoot Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
          onClick={[Function]}
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-invalid={false}
            autoCapitalize="none"
            autoComplete="off"
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused Mui-disabled Mui-disabled MuiInputBase-inputAdornedEnd"
            disabled={true}
            id="audit-log-type-details-selection"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            placeholder="-"
            required={false}
            spellCheck="false"
            type="text"
            value=""
          />
          <div
            className="MuiAutocomplete-endAdornment"
          >
            <button
              aria-label="Open"
              className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator Mui-disabled Mui-disabled"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Open"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div />
    <div
      className=""
    >
      <span>
        Filter by date
      </span>
      <ul
        className="unstyled link-list horizontal"
      >
        <li>
          <a
            className="active"
            onClick={[Function]}
          >
            Today
          </a>
        </li>
        <li>
          <a
            className=""
            onClick={[Function]}
          >
            Yesterday
          </a>
        </li>
        <li>
          <a
            className=""
            onClick={[Function]}
          >
            Last 7 days
          </a>
        </li>
        <li>
          <a
            className=""
            onClick={[Function]}
          >
            Last 30 days
          </a>
        </li>
      </ul>
    </div>
    <div
      style={
        Object {
          "gridColumnEnd": 4,
          "gridColumnStart": 2,
        }
      }
    >
      <div
        className="MuiFormControl-root MuiTextField-root"
        format="MMMM Do"
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          Object {
            "margin": 7.5,
            "marginTop": 0,
            "width": 160,
          }
        }
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink={true}
        >
          From
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={true}
            required={false}
            type="text"
            value="September 16th"
          />
        </div>
        
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root"
        format="MMMM Do"
        onClick={[Function]}
        onKeyDown={[Function]}
        style={
          Object {
            "margin": 7.5,
            "marginTop": 0,
            "width": 160,
          }
        }
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
          data-shrink={true}
        >
          To
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={true}
            required={false}
            type="text"
            value="September 16th"
          />
        </div>
        
      </div>
    </div>
  </div>
  <div
    className="flexbox"
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "flex-end",
      }
    }
  >
    <div
      className="loaderContainer shrunk"
    >
      <div
        className="loader"
      >
        <span
          className="dot dot_1"
        />
        <span
          className="dot dot_2"
        />
        <span
          className="dot dot_3"
        />
        <span
          className="dot dot_4"
        />
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "marginLeft": 15,
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Download results as csv
      </span>
    </button>
  </div>
  <div
    className="fadeIn deploy-table-contain"
  >
    <div
      className="auditlogs-list-item auditlogs-list-item-header muted"
    >
      <div
        className="columnHeader"
        onClick={[Function]}
        style={
          Object {
            "cursor": "initial",
          }
        }
      >
        User
      </div>
      <div
        className="columnHeader"
        onClick={[Function]}
        style={
          Object {
            "cursor": "initial",
          }
        }
      >
        Action
      </div>
      <div
        className="columnHeader"
        onClick={[Function]}
        style={
          Object {
            "cursor": "initial",
          }
        }
      >
        Type
      </div>
      <div
        className="columnHeader"
        onClick={[Function]}
        style={
          Object {
            "cursor": "initial",
          }
        }
      >
        Change
      </div>
      <div
        className="columnHeader"
        onClick={[Function]}
        style={
          Object {
            "cursor": "initial",
          }
        }
      >
        More details
      </div>
      <div
        className="columnHeader"
        onClick={[Function]}
        style={Object {}}
      >
        Time
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root sortIcon selected true"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h6v-2H3v2zM3 6v2h18V6H3zm0 7h12v-2H3v2z"
          />
        </svg>
      </div>
    </div>
    <div
      className="loaderContainer"
    >
      <div
        className="loader"
      >
        <span
          className="dot dot_1"
        />
        <span
          className="dot dot_2"
        />
        <span
          className="dot dot_3"
        />
        <span
          className="dot dot_4"
        />
      </div>
    </div>
    <div
      className="auditlogs-list-item"
    >
      <div>
        <div>
          string@example.com
        </div>
      </div>
      <div>
        <div
          className="uppercased"
        >
          create
        </div>
      </div>
      <div>
        <div
          className="capitalized"
        >
          user
        </div>
      </div>
      <div>
        <div>
          user@acme.com
        </div>
      </div>
      <div>
        <div
          className="capitalized"
        >
          change1
        </div>
      </div>
      <div>
        <time
          dateTime="2020-09-10T12:10:22+00:00"
          title={null}
        >
          2020-09-10 12:10
        </time>
      </div>
    </div>
    <div
      className="auditlogs-list-item"
    >
      <div>
        <div>
          string
        </div>
      </div>
      <div>
        <div
          className="uppercased"
        >
          create
        </div>
      </div>
      <div>
        <div
          className="capitalized"
        >
          deployment
        </div>
      </div>
      <div>
        <div>
          to production
        </div>
      </div>
      <div>
        <a
          href="/deployments/finished?open=true&id=string"
          onClick={[Function]}
        >
          View deployment
        </a>
      </div>
      <div>
        <time
          dateTime="2020-09-10T12:16:22+00:00"
          title={null}
        >
          2020-09-10 12:16
        </time>
      </div>
    </div>
  </div>
</div>
`;
